src/server/rdb_save.cc

```diff
@@ -907,40 +950,86 @@ error_code RdbSaver::Impl::SaveAuxFieldStrStr(string_view key, string_view val)
   return error_code{};
 }
 
+bool RdbSaver::Impl::RecordsPopper::TryPopFromQueue() {
+  if (enforce_order && !q_records.empty() && q_records.top().id == next_record_id) {
+    record_holder = std::move(const_cast<SliceSnapshot::DbRecord&>(q_records.top()));
+    q_records.pop();
+    ++next_record_id;
+    return true;
+  }
+  return false;
+}
+
+std::optional<SliceSnapshot::DbRecord> RdbSaver::Impl::RecordsPopper::Pop() {
+  return InternalPop(true);
+}
+
+std::optional<SliceSnapshot::DbRecord> RdbSaver::Impl::RecordsPopper::TryPop() {
+  return InternalPop(false);
+}
+
+std::optional<SliceSnapshot::DbRecord> RdbSaver::Impl::RecordsPopper::InternalPop(bool blocking) {
+  if (TryPopFromQueue()) {
+    return std::move(record_holder);
+  }
+
+  std::function<bool(SliceSnapshot::DbRecord&)> pop_function;
+  if (blocking) {
+    pop_function = std::bind(&SliceSnapshot::RecordChannel::Pop, channel, std::placeholders::_1);
+  } else {
+    pop_function = std::bind(&SliceSnapshot::RecordChannel::TryPop, channel, std::placeholders::_1);
+  }
```

Comment:
```c++
auto pop_fn = blocking ? &Queue::Pop : &Queue::TryPop;
while (q.*pop_fn)(channel, record_holder)) {
}
```

Reply:
Seems like a style preference, not worth mentioning.